.github/workflows/pytest.yaml

```diff
@@ -0,0 +1,41 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Run PyTest
+
+on: [push, pull_request]
+
+jobs:
+  run_pytest:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.11"
+        cache: "pip"
+
+    - name: Editable mode install
+      run: |
+        python -m pip install uv
+        uv pip install --system pytest pytest-cov pytest-html
+        uv pip install --system -e .
+    - name: Test with pytest
+      run: |
+        pytest --cov=./ --cov-report=html --html=pytest_report.html --self-contained-html
+
+    - name: Upload test report
```

Comment:
Please change to using codecov as discussed. I've added the secret to the repo.

Example code at https://github.com/lsst/daf_butler/blob/main/.github/workflows/build.yaml#L92-L98